src/__tests__/actions-test.ts

```diff
@@ -338,11 +338,11 @@ describe("actions", () => {
       }));
       fetch = fetchMock;
 
-      await actions.email(formData)(dispatch);
+      await actions.validate_email(formData)(dispatch);
       expect(dispatch.callCount).to.equal(2);
 
-      expect(dispatch.args[0][0].type).to.equal("EMAIL_REQUEST");
-      expect(dispatch.args[1][0].type).to.equal("EMAIL_SUCCESS");
+      expect(dispatch.args[0][0].type).to.equal("VALIDATE_EMAIL_REQUEST");
+      expect(dispatch.args[1][0].type).to.equal("VALIDATE_EMAIL_SUCCESS");
```

Comment:
Hmm there's no way the server can fail and you could get an error message back instead of a success?

Reply:
There is not.  The only way that something could hypothetically go wrong with validating an email is that there might not be an email address associated with the library...but in that case, the button would be disabled, so it would be impossible to get to this point in the first place.